File: backend/ecoWater/ecoWater/src/main/java/com/project/ecoWater/device/infrastructure/DeviceRequest.java
package com.project.ecoWater.device.infrastructure;

import com.project.ecoWater.device.domain.DeviceType;
import com.project.ecoWater.tank.app.TankDTO;
import com.project.ecoWater.tank.domain.Tank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class DeviceRequest

{
    String device_id;
    String device_name;
    DeviceType device_type;
    String device_location;
    String ssid;
    TankDTO tank;
}

File: frontend/Proyecto/Proyecto/app/src/main/java/com/example/proyecto/model/device/DeviceRequest.kt
package com.example.proyecto.model.device

import com.example.proyecto.model.tank.Tank

data class DeviceRequest(
    val device_id: String?=null,
    val device_name: String?=null,
    val device_type: String?=null,
    val device_location: String?=null,
    val ssid:String?=null,
    val tank: Tank?=null
)